nwb_conversion_tools/datainterfaces/ecephys/baselfpextractorinterface.py

```diff
@@ -1,25 +1,93 @@
 """Authors: Cody Baker and Ben Dichter."""
 from typing import Optional, Union
 from pathlib import Path
+import numpy as np 
 
 from pynwb import NWBFile
+from pynwb.device import Device
+from pynwb.ecephys import ElectrodeGroup
 
 from .baserecordingextractorinterface import BaseRecordingExtractorInterface
 from ...utils.spike_interface import write_recording
+from ...utils.json_schema import (
+    get_schema_from_hdmf_class,
+    get_base_schema
+)
 
 OptionalPathType = Optional[Union[str, Path]]
 
 
 class BaseLFPExtractorInterface(BaseRecordingExtractorInterface):
     """Primary class for all LFP data interfaces."""
 
+    def get_metadata_schema(self):
+        """Compile metadata schema for the RecordingExtractor."""
+        metadata_schema = super().get_metadata_schema()
+
+        # Initiate Ecephys metadata
+        metadata_schema['properties']['Ecephys'] = get_base_schema(tag='Ecephys')
+        metadata_schema['properties']['Ecephys']['required'] = ['Device', 'ElectrodeGroup']
+        metadata_schema['properties']['Ecephys']['properties'] = dict(
+            Device=dict(
+                type="array",
+                minItems=1,
+                items={"$ref": "#/properties/Ecephys/properties/definitions/Device"}
+            ),
+            ElectrodeGroup=dict(
+                type="array",
+                minItems=1,
+                items={"$ref": "#/properties/Ecephys/properties/definitions/ElectrodeGroup"}
+            ),
+            Electrodes=dict(
+                type="array",
+                minItems=0,
+                renderForm=False,
+                items={"$ref": "#/properties/Ecephys/properties/definitions/Electrodes"}
+            ),
+        )
+        # Schema definition for arrays
+        metadata_schema['properties']['Ecephys']['properties']["definitions"] = dict(
+            Device=get_schema_from_hdmf_class(Device),
+            ElectrodeGroup=get_schema_from_hdmf_class(ElectrodeGroup),
+            Electrodes=dict(
+                type="object",
+                additionalProperties=False,
+                required=["name"],
+                properties=dict(
+                    name=dict(
+                        type="string",
+                        description="name of this electrodes column"
+                    ),
+                    description=dict(
+                        type="string",
+                        description="description of this electrodes column"
+                    )
+                )
+            )
+        )
+        return metadata_schema
+
```

Comment:
is this not a repetition from the base class `BaseRecordingExtractorInterface`

Reply:
@Saksham20 is correct - if one follows the inheritance structure it is duplicated from its super